### PR TITLE
use separate track for 3.x

### PR DESCRIPTION
**Pull request by rgildein**
Change release workflow to release on push to 3.x branch and push it to the 3.x/edge instead of latest/edge.

### CODE REVIEW

**Review by jneo8 — CHANGES_REQUESTED** [review]
Can you explain more about the `3.x` track?

Maybe add some description in README.

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: Publish snap
 
 on:
   push:
-    branches: [ main ]
+    branches: [ '*.x' ]
```
Comment (chanchiwai-ray) [inline .github/workflows/release.yaml:5]: Why use `*.x` instead of more explicit `3.x` ?

Comment (rgildein) [reply .github/workflows/release.yaml:5]: Good point, we can use `3.x` directly.

**Comment by rgildein** [comment]
> Can you explain more about the `3.x` track?
> 
> Maybe add some description in README.

What do you want to explain?

**Comment by jneo8** [comment]
> What do you want to explain?

When to use 3.x and when to use master. I think we should have that on the README.md.

**Comment by rgildein** [comment]
> > What do you want to explain?
> 
> When to use 3.x and when to use main. I think we should have that on the README.md. And the reason why we separate the track.

This can be done by another PR, my intention with this initiative was to help support for 3.x Juju.

**Comment by rgildein** [comment]
Closing, since there should be proper task to do:
- add track to snapcraft.io
- add some note to document
- update workflow to publish to proper track/risk

**Pull request closed without merging**